### PR TITLE
eos_route_maps resource module

### PR DESCRIPTION
**Pull request by GomathiselviS**
##### SUMMARY

Added eos_route_maps resource module.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

eos_route_maps
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by GomathiselviS** [comment]
recheck

File: plugins/modules/eos_route_maps.py [hunk 1]
```diff
@@ -0,0 +1,1404 @@
...
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: eos_route_maps
+short_description: Manages Route Maps resource module
+description: This module configures and manages the attributes of Route Mapd on Arista
+  EOS platforms.
+version_added: 2.0.0
```
Comment (justjais) [inline plugins/modules/eos_route_maps.py:38]: version_added needs to be updated

File: plugins/modules/eos_route_maps.py [hunk 2]
```diff
@@ -0,0 +1,1404 @@
...
+notes:
+- Tested against Arista EOS 4.23.0F
+- This module works with connection C(network_cli). See the L(EOS Platform Options,eos_platform_options).
+options:
+    config:
+      description: A list of route-map options
+      type: list
+      elements: dict
+      suboptions:
+        route_map_name:
```
Comment (justjais) [inline plugins/modules/eos_route_maps.py:49]: can we change to `route_map` instead

File: plugins/modules/eos_route_maps.py [hunk 3]
```diff
@@ -0,0 +1,1404 @@
...
+                  description: Set as-path.
+                  type: dict
+                  suboptions:
+                    match:
+                      description: Match the entire as-path.
+                      type: dict
+                      suboptions:
+                        as_number:
+                          description: as number to use (includes auto;in csv format)
+                          type: str
```
Comment (justjais) [inline plugins/modules/eos_route_maps.py:111]: here, `as_number` should be list of str as it supports list

File: plugins/modules/eos_route_maps.py [hunk 4]
```diff
@@ -0,0 +1,1404 @@
...
+                        none:
+                          description: Remove matching AS numbers
+                          type: bool
+                    prepend:
+                      description: Prepend to the as-path.
+                      type: dict
+                      suboptions:
+                        as_number:
```
Comment (justjais) [inline plugins/modules/eos_route_maps.py:119]: here, `as_number` should be list of str as it supports list

File: plugins/modules/eos_route_maps.py [hunk 5]
```diff
@@ -0,0 +1,1404 @@
...
+                          description: as number to prepend (includes auto;in csv format)
+                          type: str
+                        last_as:
+                          description: The number of times to prepend the last AS number.
+                          type: int
+                bgp:
+                  description: BGP AS path multipath weight.
+                  type: int
+                community_attributes:
```
Comment (justjais) [inline plugins/modules/eos_route_maps.py:128]:
```suggestion
                community:
```
as that's how it's on the box

File: plugins/modules/eos_route_maps.py [hunk 6]
```diff
@@ -0,0 +1,1404 @@
...
+                    graceful_shutdown:
+                      description: Graceful shutdown
+                      type: bool
+                    community:
+                      description: community attributes.
+                      type: dict
+                      suboptions:
+                        number:
+                          description: community number (in csv format).
+                          type: str
```
Comment (justjais) [inline plugins/modules/eos_route_maps.py:141]: community number should be list of number, correct?

File: plugins/modules/eos_route_maps.py [hunk 7]
```diff
@@ -0,0 +1,1404 @@
...
+                  description: Route metric type.
+                  type: str
+                  choices: ["type-1", "type-2"]
+                nexthop:
+                  description: Route next hop.
+                  type: dict
+                  suboptions:
+                    value:
+                      description: IGP metric value.
+                      type: int
```
Comment (justjais) [inline plugins/modules/eos_route_maps.py:261]: can u plz check if this also supports list of int

File: tests/unit/modules/network/eos/test_eos_route_maps.py [hunk 8]
```diff
@@ -0,0 +1,481 @@
...
+        if filename is None:
+            filename = "eos_route_maps_config.cfg"
+
+        def load_from_file(*args, **kwargs):
+            output = load_fixture(filename)
+            return output
+
+        self.execute_show_command.side_effect = load_from_file
+
+    def test_eos_route_maps_merged_idempotent(self):
```
Comment (justjais) [inline tests/unit/modules/network/eos/test_eos_route_maps.py:51]: wanted to check if you covered all the params supported under route_maps RM via unit TC, as now we decided to tc of all module params which can give best possible coverage.

**Review by justjais — CHANGES_REQUESTED** [review]

File: plugins/modules/eos_route_maps.py (hunk 5 above)
Comment (GomathiselviS) [reply plugins/modules/eos_route_maps.py:128]: There are some attributes to be mentioned after we specify the community number and some attributes at the community number level. So I have got a separate key named community (dict) , under community_attributes.

File: plugins/modules/eos_route_maps.py (hunk 6 above)
Comment (GomathiselviS) [reply plugins/modules/eos_route_maps.py:141]: I am getting the list in csv format, which is processed inside the rm_template.

File: plugins/modules/eos_route_maps.py (hunk 7 above)
Comment (GomathiselviS) [reply plugins/modules/eos_route_maps.py:261]: It takes a single integer.

File: tests/unit/modules/network/eos/test_eos_route_maps.py (hunk 8 above)
Comment (GomathiselviS) [reply tests/unit/modules/network/eos/test_eos_route_maps.py:51]: The coverage for argspecs for this module is > 80%. Having said that, I have submitted a separate issue to increase the test coverage in eos modules. I will be working on this module along with other modules, to close the issue. (https://github.com/ansible-collections/arista.eos/issues/204)

File: plugins/modules/eos_route_maps.py (hunk 3 above)
Comment (GomathiselviS) [reply plugins/modules/eos_route_maps.py:111]: I obtain the value in csv format, and handle it inside rm_templates.

File: plugins/modules/eos_route_maps.py (hunk 4 above)
Comment (GomathiselviS) [reply plugins/modules/eos_route_maps.py:119]: I obtain the value in csv format, and handle it inside rm_templates.

File: plugins/modules/eos_route_maps.py [hunk 9]
```diff
@@ -0,0 +1,1404 @@
...
+                  description: if True, overwrite existing config.
+                  type: bool
+            action:
+              description: Action for matching routes
+              type: str
+              choices: ["deny", "permit"]
+            sequence:
+              description: Index in the sequence.
+              type: int
+            continue_map:
```
Comment (NilashishC) [inline plugins/modules/eos_route_maps.py:81]: If this points to a sequence and not another route_map, can we change this to `continue_sequence`?

**Comment by GomathiselviS** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Review by rohitthakur2590 — APPROVED** [review]
LGTM!

**Pull request merged**